app/assets/javascripts/discourse/app/lib/utilities.js

```diff
@@ -632,25 +632,30 @@ export function getCaretPosition(element, options) {
  * Generate markdown table from an array of objects
  * Inspired by https://github.com/Ygilany/array-to-table
  *
- * @param  {Array} array       Array of objects
- * @param  {Array} columns     Column headings
- * @param  {String} colPrefix  Table column prefix
+ * @param  {Array<Record<string, string | undefined>>}          array       Array of objects
+ * @param  {String[]}                                           columns     Column headings
+ * @param  {String}                                             colPrefix   Table column prefix
+ * @param  {("left" | "center" | "right" | null)[] | undefined} alignments  Table alignments
  *
  * @return {String} Markdown table
  */
-export function arrayToTable(array, cols, colPrefix = "col") {
+export function arrayToTable(array, cols, colPrefix = "col", alignments) {
```

Comment:
Should we have the "mandatory" `alignments` argument on the "left" of the "optional" `colPrefix` argument? 🤔 I know in JS it doesn't matter than much, but at least it's consistent with how it's done in Ruby?

Reply:
actually the `alignments` is also optional :thinking: It’s just that in JS we don’t used to write such: (In JS we don’t have a `?` to explicitly declare it as optional)

```javascript
export function arrayToTable(array, cols, colPrefix = "col", alignments = undefined) 
```

It should actually be written like this in TS:

```typescript
export function arrayToTable(array: object[], cols: string[], colPrefix = "col", alignments?) 
```